File: 2717. Semi-Ordered Permutation/solution.py
class Solution(object):
  def semiOrderedPermutation(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    indexOne = 0
    indexN = len(nums) - 1
    for index, num in enumerate(nums):
      if num == 1:
        indexOne = index
      if num == len(nums):
        indexN = index
    
    total = (indexOne - 0) + (len(nums) - 1 - indexN)
    if indexN < indexOne:
      total -= 1
    return total

File: 2717. Semi-Ordered Permutation/solution.java
class Solution {
  public int semiOrderedPermutation(int[] nums) {
    int indexOne = 0;
    int indexN = nums.length - 1;
    for (int index = 0; index < nums.length; index++){
      int num = nums[index];
      if (num == 1){
        indexOne = index;
      }
      if (num == nums.length){
        indexN = index;
      }
    }
    
    int total = (indexOne - 0) + (nums.length - 1 - indexN);
    if (indexN < indexOne){
      total -= 1;
    }
    return total;
  }
}